extensions/shared/premium-blocks/with-upgrade-banner.jsx

```diff
@@ -1,40 +1,61 @@
 /**
  * External dependencies
  */
+import classNames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
 import { createHigherOrderComponent } from '@wordpress/compose';
-import { Fragment } from '@wordpress/element';
+import { useState, useEffect } from '@wordpress/element';
 import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
-import { isUpgradable } from '../plan-utils';
+import { isStillUsableWithFreePlan, isUpgradable } from '../plan-utils';
 import UpgradePlanBanner from './upgrade-plan-banner';
+import { PremiumBlockProvider } from './components';
 
 export default createHigherOrderComponent(
 	BlockListBlock => props => {
 		if ( ! isUpgradable( props?.name ) ) {
 			return <BlockListBlock { ...props } />;
 		}
 
+		const isDualMode = isStillUsableWithFreePlan( props?.name );
+
+		const [ isVisible, setIsVisible ] = useState( ! isDualMode );
+
+		// Hide Banner when block changes its attributes (dual Mode).
+		useEffect( () =>
+			setIsVisible( ! isDualMode )
+			, [ props.attributes, setIsVisible, isDualMode ]
```

Comment:
This line formatting is a bit confusing, could that be changed to

```suggestion
		useEffect(
			() => setIsVisible( ! isDualMode ),
			[ props.attributes, setIsVisible, isDualMode ]
```

Reply:
just fyi, we are going to make a whole tidying code once this PR gets merged with the feature branch.

Reply:
sorry I didn't realize it was a commit suggestion.